src/request.rs

```diff
@@ -0,0 +1,80 @@
+use super::*;
+use std::collections::HashMap;
+use url::Url;
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct Asset {
+    /// The request Url
+    pub url: Url,
+    /// Name of the asset, usually the filename
+    pub name: Option<String>,
+    /// Description of the asset
+    pub description: Option<String>,
+    /// Additional headers to send with the request
+    pub headers: HashMap<String, String>,
+    /// Request type
+    pub request_type: RequestType,
+    /// Hashes to check
+    pub hash: Option<AssetHash>,
+    /// Optional size of file in bytes
+    pub size: Option<usize>,
+}
+
+impl Asset {
```

Comment:
That sounds like a good place to put a builder in
Either make it manually, or there's a whole lot of procedural macros that can do the job

Reply:
I've never heard of that, but yeah a better implementation of the contructors for this struct is sorely needed